src/components/EmailNotifications/VerifyEmailAddress.js

```diff
@@ -0,0 +1,258 @@
+                            css={`
+                              opacity: 0;
+                            `}
+                          />
+                        )
+                      }
+                      adornmentPosition="end"
+                      type="email"
+                      wide
+                      onChange={handleEmailAddressChange}
+                      placeholder="email@address.com"
+                      onBlur={handleEmailAddressBlur}
+                    />
+                    {emailInvalid && (
+                      <p
+                        css={`
+                          color: ${theme.negative};
+                          ${textStyle('body4')};
+                          float: left;
+                          margin-top: ${0.5 * GU}px;
+                        `}
+                      >
+                        Please enter a valid email address.
+                      </p>
+                    )}
+                  </Field>
+                </div>
+                <LegalTermsAndPolicy
+                  termsAccepted={termsAccepted}
+                  setTermsAccepted={setTermsAccepted}
+                  theme={theme}
+                />
+                <div
+                  css={`
+                    display: flex;
+                    justify-content: space-between;
+                    flex-direction: ${compactMode ? 'column' : 'row'};
+                    width: 100%;
+                    margin-bottom: ${1.5 * GU}px;
+                    margin-top: ${3 * GU}px;
+                  `}
+                >
+                  <ActionButtons
+                    compactMode={compactMode}
+                    onClick={onDeleteEmail}
+                    width={buttonWidth}
+                  >
+                    Delete email
+                  </ActionButtons>
+                  <ActionButtons
+                    compactMode={compactMode}
+                    mode="strong"
+                    disabled={emailInvalid || !termsAccepted}
+                    onClick={handleSubscribe}
+                    width={buttonWidth}
+                  >
+                    Send verification email
+                  </ActionButtons>
+                </div>
+              </>
+            )
+          }
+          return (
+            <Button
+              css={`
+                margin-top: ${3 * GU}px;
+              `}
+              mode="strong"
+              onClick={handleReSendEmail}
+            >
+              Resend verification email
+            </Button>
+          )
+        })()}
+      </div>
+    </div>
+  )
+})
+
+function LegalTermsAndPolicy({ termsAccepted, setTermsAccepted, theme }) {
+  return (
+    <div
+      css={`
+        margin-top: ${3 * GU}px;
+      `}
+    >
+      <div
+        css={`
+          display: flex;
+        `}
+      >
+        <Checkbox
+          checked={termsAccepted}
+          onChange={checked => setTermsAccepted(checked)}
```

Comment:
```suggestion
          onChange={onTermsAccepted}
```

Rather than directly exposing the `setState()` function, we should be generic and accept a handler for when the `onChange()` is triggered.